decidim-admin/app/controllers/decidim/admin/resource_permissions_controller.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Admin
+    # Controller that allows managing resource permissions.
+    #
+    class ResourcePermissionsController < Decidim::Admin::ApplicationController
+      helper Decidim::ResourceHelper
+
+      helper_method :authorizations, :other_authorizations_for, :resource_params, :resource
+
+      def edit
+        @permissions_form = PermissionsForm.new(
+          permissions: permission_forms
+        )
+      end
+
+      def update
+        @permissions_form = PermissionsForm.from_params(params).with_context(current_organization: current_organization)
+
+        UpdateResourcePermissions.call(@permissions_form, resource) do
+          on(:ok) do
+            flash[:notice] = t("resource_permissions.update.success", scope: "decidim.admin")
+            redirect_to return_path
+          end
+
+          on(:invalid) do
+            render action: :edit
+          end
+        end
+      end
+
+      private
+
+      def return_path
+        send("#{resource_symbol.to_s.pluralize}_path")
```

Comment:
Could we use the `ResourceLocator` instead of this? I'm not comfortable using `send` here to build a route path

Reply:
This path is the admin index of the resource. The `ResourceLocatorPresenter` works with resources associated with components and participatory spaces. In this PR the controller is used to manage permissions of a `InitiativesType`. The ResourceLocator for InitiativesType and index path expects a `mounted_engine` and a `mounted_params`. Even if these methods are defined in the model the engine router produces an error trying to get `initiatives_types_path` from `decidim_initiatives` because the correct engine is the `decidim_admin_initiatives`, not the front one. I've added an `admin_index` method for `ResourceLocatorPresenter` making use of `admin_mounted_engine` instead of `mounted_engine`